### DISCUSSION TITLE
session file

### DISCUSSION THREAD

**Discussion by stusemple** [JoTec2002/InstagramToMealie#1] [Ideas]
First of all, thanks so much for making this. It looks amazing and ideal for my use case (partner who loves to find recipes from instagram). Is there any way you can make the session file generation easier? I am struggling to generate my own

**Comment by JoTec2002** [comment]
I initial tried using the username + password + TOTP Version of the Instagram Scraper Library, but that Feature is horribly buggy.

If you're using Windows i found Running the Script relativly straight forward.

1. Install Firefox 
2. Log into Instagram in Firefox
3. run the Python Script (I think if you don't have Python installed on your Windows Maschine it will take you to the Windows Store from where it could be installed it very easily. Linux should already have it installed)

Little disclaimer i didn't Write this Script it comes from the Instascraper Library.

If this doesn't help you I'm happy to help you further, if you could explain your issues further.

**Reply by JoTec2002** [reply]
I updated the Docker Container so that you can try the insta login with username+password but as explained in the updated Readme I wouldn't recommand it.

If you need help with creating the sessionfile I'm still happy to help yoz.

**Reply by stusemple** [reply]
Amazing! I tried using the method in the docker container itself but as you suggested it didn’t work. I had a bit of a saga installing python on my laptop and generating the session file but I sorted it this morning and just imported my first recipe direct from instagram. What a brilliant bit of work, thanks so much.

**Comment by stanthewizzard** [comment]
sorry I did (on windows too) created the session file
then copied to session-file folder on my docker (the folder is meali_insta and inside it session-file)
mounted on compose

`Failed to get a session file or Instagram password. Provide a valid session file or set INSTA_PWD in environment`

I'm a bit lost
Thanks

**Reply by stanthewizzard** [reply]
without 2FA it's ok
I can log
I don't understand how to keep 2FA ?
Do you think there is a way to have translation  ? AI maybe ?
THANKS

**Reply by JoTec2002** [reply]
I myself use 2FA for Instagram should work without problems.
When you logged in into Firefox you can maybe try and set "remember me" and "mark as secure device" when logging in to Instagram maybe this helps.
So just to clarify you can login via the session file, when 2FA is disabled but not when 2FA is enabled?

**Reply by stanthewizzard** [reply]
Not at all with the file
Putting login /password without 2FA works
Your scrapper is awesome btw

**Reply by JoTec2002** [reply]
Login via session file should definitely work. You just need to mount the session file to the docker container via compose. Don't map the whole folder just the file. 
`- "./session-file:/app/session-file"`

**Reply by stanthewizzard** [reply]
I mapped the whole folder. Trying your compose

**Reply by stanthewizzard** [reply]
mapping only the file WORKS :)

**Reply by stanthewizzard** [reply]
Now that it is working properly your container is just awesome !!!
I need to find a way to translate what is in english :)
thank you thank you thank you

PS: if you have any plan for tiktok too :)

**Reply by JoTec2002** [reply]
That's nice to hear.

I personally always "improve" the recipe after import (for example: converting from string ingredients to ingredients with amounts) while doing this I just use Google translator and copy paste.

**Reply by stanthewizzard** [reply]
yes but I need to have some automation (insta is filled with more than 1200 recipes)
That a question for .... chatGPT :) (on how to use meali api with ollama whatever model to translate)
